core/src/clusterization/measurement_creation.cpp

```diff
@@ -36,25 +36,24 @@ measurement_creation::output_type measurement_creation::operator()(
         // [2] The Art of Computer Programming, Donald E. Knuth, second
         //     edition, chapter 4.2.2.
 
-        // Get the cell module
-        const auto &module_link = clusters.at(i).header;
-        const auto &module = cells.at(module_link).header;
-
-        if (result.size() == 0 ||
-            result.get_headers().back().module != module.module) {
-            result.push_back(cell_module(module),
-                             measurement_collection_types::host(&(m_mr.get())));
-        }
-
         // Get the cluster.
         cluster_container_types::host::item_vector::const_reference cluster =
             clusters.at(i).items;
 
+        if (cluster.size() == 0) {
+            // this should never happen, but included for safekeeping.
```

Comment:
:thinking: If it should really never happen, then let's put an assertion here. I.e.

```c++
assert(cluster.size() > 0);
```